### PR TITLE
Feature/install paths

### PR DESCRIPTION
**Pull request by BorjaOuterelo**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/50/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/51/)

File: CMakeLists.txt [hunk 1]
```diff
@@ -395,12 +398,18 @@ install(
...
-            ${CMAKE_BINARY_DIR}/temp_install/
-        DESTINATION
-            ${CMAKE_INSTALL_PREFIX}
+   foreach(dependency ${_deps})
+        list(GET dependency 0 _name)
+        list(GET dependency 1 _version)
+        install(
+            DIRECTORY
+                ${CMAKE_BINARY_DIR}/temp_install/${_name}-${_version}
+            DESTINATION .
```
Comment (julianbermudez) [inline CMakeLists.txt]: I prefer the `${CMAKE_INSTALL_PREFIX}`, what do you think?

File: CMakeLists.txt [hunk 2]
```diff
@@ -183,6 +172,20 @@ set(SRCS
...
+set(BIN_INSTALL_DIR     ${PROJECT_NAME}-${PROJECT_VERSION}/${CMAKE_INSTALL_BINDIR}     CACHE PATH "Installation directory for binaries")
+set(INCLUDE_INSTALL_DIR ${PROJECT_NAME}-${PROJECT_VERSION}/${CMAKE_INSTALL_INCLUDEDIR} CACHE PATH "Installation directory for C headers")
+set(LIB_INSTALL_DIR     ${PROJECT_NAME}-${PROJECT_VERSION}/${CMAKE_INSTALL_LIBDIR}     CACHE PATH "Installation directory for libraries")
+set(DATA_INSTALL_DIR    ${PROJECT_NAME}-${PROJECT_VERSION}/${CMAKE_INSTALL_DATADIR}    CACHE PATH "Installation directory for data")
+if(WIN32)
+    set(LICENSE_INSTALL_DIR . CACHE PATH "Installation directory for licenses")
+else()
+    set(LICENSE_INSTALL_DIR ${DATA_INSTALL_DIR}/${PROJECT_NAME} CACHE PATH "Installation directory for licenses")
+endif()
+
```
Comment (julianbermudez) [inline CMakeLists.txt]: Why do you move it?

File: CMakeLists.txt [hunk 3]
```diff
@@ -237,7 +240,7 @@ target_include_directories(${PROJECT_NAME}
     PUBLIC
         $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
         $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
-        $<INSTALL_INTERFACE:include>
+        $<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>
```
Comment (julianbermudez) [inline CMakeLists.txt]: Nice :wink:

File: cmake/SuperBuild.cmake [hunk 4]
```diff
@@ -23,15 +23,15 @@ enable_language(CXX)
...
         GIT_REPOSITORY
             https://github.com/eProsima/Micro-CDR.git
         GIT_TAG
             ${_microcdr_tag}
         PREFIX
-            ${PROJECT_BINARY_DIR}/ucdr
+            ${PROJECT_BINARY_DIR}/microcdr
         INSTALL_DIR
-            ${PROJECT_BINARY_DIR}/temp_install
+            ${PROJECT_BINARY_DIR}/temp_install/microcdr-${_microcdr_version}
```
Comment (julianbermudez) [inline cmake/SuperBuild.cmake]: I like this new approach

File: CMakeLists.txt (hunk 2 above)
Comment (BorjaOuterelo) [reply CMakeLists.txt]: `PROJECT_NAME` and `PROJECT_VERSION` are populated in the call to `project`.

File: CMakeLists.txt (hunk 1 above)
Comment (BorjaOuterelo) [reply CMakeLists.txt]: I am not sure if using `${CMAKE_INSTALL_PREFIX}`  invalidates de use of `DESTDIR` we should check.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/56/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/57/)

**Pull request merged**